### ISSUE TITLE
Сделать второй светодиод (как в SportIdent)

### ISSUE THREAD

**Issue by mforis** [alexandervolikov/sportiduino#39]
Есть предложение добавить в прошивку возможность подключения второго светодиода, который будет загораться с задержкой (к примеру 0,5 секунды) и только в рабочем режиме. Например, при программировании базовых станций он загораться не будет.

**Comment by alexandervolikov** [comment]
Добавить можно. Только не понимаю какие это удобства даст. Отличать сигнал при программировании от отметки для того, чтобы убедиться в прохождении процедуры?

**Comment by mforis** [comment]
Не совсем, участники соревнований иногда жалуются, что при прохождении отметки не слышно звукового сигнала (в городской черте) и не видно светодиода (за счет того, что он закрывается чипом), ну и в принципе,  то что вы сказали тоже. Просто, по-моему у МК остаются свободные выводы, так что если в это не сложно воплатить, то думаю, это было бы полезно.

**Comment by alexandervolikov** [comment]
Понятно.
А если просто сделать мигание тем же светодиодом через данный интервал, но без звукового сигнала?

**Comment by mforis** [comment]
Хм... Просто, не будут ли накладываться сигналы друг на друга в случае большого количества спортсменов на КП?

**Comment by alexandervolikov** [comment]
Проблемма ардуино ещё в том, что он однозадачный. Не будет обрабатывать следующего чипа пока не проделает всю процедуру отметки, в том числе мигнёт вторым светодиодом. Поэтому, думаю, просто добавить второй световой сигнал спуста 0.5-1 секунду будет равнозначно добавлению второго светодиода в этом контексте. Я попробую как оно будет. Просто делать дополнительный светодиод - это терять полноценную совместимость со станциями, которые уже наделаны.

**Comment by mforis** [comment]
Александр, обратил внимание, на то, что когда напряжение на батарейках садиться до 3.8 вольт база перестает работать или в лучшем случае, очень плохо считывает чипы.

**Comment by alexandervolikov** [comment]
Любопытно. Я вроде тестировал, работало вплоть до 2.8 вольт. Надо будет ещё посмотреть. Что 3.8 вольт весьма странно, там стоит линейный стабилизатор, который всё понижает до 3.3 вольт, хоть 3.8, хоть 5 вольт. Может на холоде батарейки просели?

**Comment by mforis** [comment]
@alexandervolikov Так, это не первый раз, симптомы такие: станция перестает реагировать на чип, или выходит из спящего режима, но после попытки отметки входит в какой-то цикл (раз в несколько секунд тускло загорается светодиод и иногда тихий звуковой сигнал).

**Comment by alexandervolikov** [comment]
дублирующий световой сигнал добавил. Получилось вроде получше. С питанием пока воспроизвести баг не удалось.

**Comment by mforis** [comment]
Работают ли базовые станции с этой прошивкой, вместе с базами с прошивкой 1.3?

**Comment by alexandervolikov** [comment]
Да, работают. Но полная совместимость утерена в виду изменения байта настроек станции. В частности, изменено назначение битов 4-7. Если раньше настройками станций не пользовались, то разницы никакой. Если ранее использовали настройки, то с новой прошивкой нужно будет обновить их.

**Comment by mforis** [comment]
Понял, спасибо! Сегодня попробую запитать базовую станцию от блока питания и посмотрю, как она себя будет вести, при разных напряжениях...

**Comment by alexandervolikov** [comment]
Потестировал несколько базовых станций, работают вплоть до 2.8 вольт на батарейках. Возможно, в Вашем случае какие-то проблемы со стабилизатором.

**Comment by mforis** [comment]
У меня от БП тоже работало вплоть до 2.8, а от батареек, при 3.4-3.8 базовые станции работают не стабильно, то считыватель отрубает, то база в циклическую загрузку уходит...

**Comment by alexandervolikov** [comment]
Значит, дело в батарейках. Стоит учесть, что под нагрузкой напряжение на них может довольно значительно упасть. Я тестировал именно с батареек, но они у меня довольно хорошие - Duracel Industrial.

**Comment by mforis** [comment]
@alexandervolikov Ну всё, значит нашли в чем проблема... Возможно в будущем, переведу свои станции на АКБ.

**Comment by mforis** [comment]
Нашел баг в прошивке 1.4.1: после прохождения отметки станция перезагружается

**Comment by alexandervolikov** [comment]
Странно. Но может быть, я её толком не тестировал. Посмотрю в чем может быть дело

**Comment by mforis** [comment]
Такой баг только в прошивке 1.4.1, в 1.4.0 станция не перезагружается, но при попытке уменьшить мощность считывателя, станция перестает видеть чипы

**Comment by alexandervolikov** [comment]
Спасибо, посмотрю.
Думаю, я сделаю так, чтобы мощность антены можно было настроить только при прошивке.

**Comment by alexandervolikov** [comment]
Нашёл возможную проблему перезагрузки. Исправил. 
Также убрал настройки мощности антены внутрь прошивки, так что, если нужна данная настройка, то нужно прописывать перед прошивкой конкретной станции.

**Comment by mforis** [comment]
Есть идея, добавить в базовую станцию дисплей, не знаю на сколько это будет полезно. Но на нем могла бы отображаться информации о времени, номере КП и о последнем отметившимся участнике.

**Comment by mforis** [comment]
![](https://i.imgur.com/Zsomuo8.png)

**Comment by halny123** [comment]
Думаю дисплей в базовых станциях излишен, т.к. хрупко и в принципе толку от него не много.
А в мастер станции (если сделать автономное питание) можно выводить время прохождения  дистанции и отметки, для тренировок и быстрой проверки результатов.

В своих станциях добавил геркон, который теперь будит и усыпляет станции. Оказалось очень удобно не нужно ждать 25с чтобы разбудить, и можно усыпить станции в лесу,  при съеме КП, без компьютера. Плюс синхронизация времени через радиомодуль порадовала, время на всех станциях синхронизируется точно по нажатию на одну кнопку.
